pandas/tseries/resample.py

```diff
@@ -743,23 +742,24 @@ def _upsample(self, method, limit=None):
 
         ax = self.ax
         obj = self.obj
-
         new_index = self._get_new_index()
+
         if len(new_index) == 0:
             return self._wrap_result(new_index)
 
-        if not is_superperiod(ax.freq, self.freq):
-            return self.asfreq()
-
         # Start vs. end of period
         memb = ax.asfreq(self.freq, how=self.convention)
 
         # Get the fill indexer
         indexer = memb.get_indexer(new_index, method=method, limit=limit)
-        return self._wrap_result(_take_new_index(obj,
-                                                 indexer,
-                                                 new_index,
-                                                 axis=self.axis))
+        return self._wrap_result(_take_new_index(
+            obj, indexer, new_index, axis=self.axis))
+
+    def _groupby_and_aggregate(self, grouper, how, *args, **kwargs):
```

Comment:
I'm not sure that makes it that much clearer. I also imagine a lot of this will go away when `PeriodIndex` becomes an `IntervalIndex`, so there's a high discount rate on any cleaning here.

Are you OK to leave as-is for the moment?